File: 백준/Silver/10816. 숫자 카드 2/숫자 카드 2.js
const input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n");

const N = input.shift();
const cards = input.shift().split(" ").map(Number);
const M = input.shift();
const integer = input[0].split(" ").map(Number);
let result = [];

let map = new Map();
cards.forEach((el) => {
    if (map.has(el)) { map.set(el, map.get(el) + 1) } else { map.set(el, 1) }
})

integer.forEach((el) => {
    if (map.has(el)) { result.push(map.get(el)) } else { result.push(0) }
})

console.log(result.join(" "));

File: 백준/Silver/10816. 숫자 카드 2/숫자 카드 2.py
import sys

N = int(sys.stdin.readline().strip())
N_s = list(map(int, sys.stdin.readline().strip().split()))
M = int(sys.stdin.readline().strip())
M_s = list(map(int, sys.stdin.readline().strip().split()))

n_dic = {}

for n in N_s:
  if n in n_dic:
    n_dic[n] += 1
  else:
    n_dic[n] = 1

for m in M_s:
  if m in n_dic:
    print(n_dic[m], end=' ')
  else:
     print(0, end=' ')